test/qa/lib/allowlist.py

```diff
@@ -0,0 +1,151 @@
+import copy
+import ipaddress
+import sh
+
+import lib
+from lib import daemon, firewall
+
+
+MSG_ALLOWLIST_ADD_SUBNET_SUCCESS = "Subnet %s is allowlisted successfully."
+MSG_ALLOWLIST_REMOVE_SUBNET_SUCCESS = "Subnet %s is removed from the allowlist successfully."
+
+MSG_ALLOWLIST_ADD_PORT_SUCCESS = "Port %s (%s) is allowlisted successfully."
+MSG_ALLOWLIST_REMOVE_PORT_SUCCESS = "Port %s (%s) is removed from the allowlist successfully."
+
+MSG_ALLOWLIST_ADD_PORT_RANGE_SUCCESS = "Ports %s (%s) are allowlisted successfully."
+MSG_ALLOWLIST_REMOVE_PORT_RANGE_SUCCESS = "Ports %s (%s) are removed from the allowlist successfully."
+
+
+def _is_private_subnet(subnet_str):
+    private_networks = [
+        ipaddress.IPv4Network('10.0.0.0/8'),
+        ipaddress.IPv4Network('172.16.0.0/12'),
+        ipaddress.IPv4Network('192.168.0.0/16')
+    ]
+
+    try:
+        subnet = ipaddress.IPv4Network(subnet_str)
+        for private_net in private_networks:
+            if subnet.overlaps(private_net):
+                return True
+        return False
+    except ValueError:
+        # Handle the case where an invalid subnet string is provided
+        return False
+
+
+def add_ports_to_allowlist(ports_list: list[lib.Port], allowlist_alias="allowlist") -> bool:
+    _ports_list = copy.deepcopy(ports_list) # We don't want to alter value later in original list
```

Comment:
There is no need to copy the list at all, because you reference to `port.value` only in the single iteration. Just make:
``` Python
for port in ports_list:
    if ":" in port.value:
        # Port range
        range_start, range_end = port.value.split(":")
        port_value = port.value.replace(":", " - ")
        ...
    else:
        # Single port
        port_value = port.value
        ...
```
and use `port_value` instead `port.value` everywhere down there in the loop.